packages/web3-utils/src/validation.ts

```diff
@@ -109,56 +109,34 @@ export const isTopicInBloom = isTopicInBloomValidator;
  * Returns -1 if a \< b, returns 1 if a \> b and returns 0 if a == b
  */
 export const compareBlockNumbers = (blockA: BlockNumberOrTag, blockB: BlockNumberOrTag) => {
-	// string validation
-	if (blockA === 'genesis' || blockB === 'genesis')
-		throw new InvalidBlockError('Genesis tag not supported'); // for more specific error message
-	if (typeof blockA === 'string' && !isBlockTag(blockA)) throw new InvalidBlockError(blockA);
-	if (typeof blockB === 'string' && !isBlockTag(blockB)) throw new InvalidBlockError(blockB);
-
-	// Increasing order:  earliest, finalized , safe, latest, pending
-	// safe vs block-num cant be compared as block number provided can be on left or right side of safe tag, until safe tag block number is extracted and compared
+	const isABlockTag = typeof blockA === 'string' && isBlockTag(blockA);
+	const isBBlockTag = typeof blockB === 'string' && isBlockTag(blockB);
 
 	if (
 		blockA === blockB ||
-		((blockA === 'earliest' || blockA === 0) && (blockB === 'earliest' || blockB === 0))
+		((blockA === 'earliest' || blockA === 0) && (blockB === 'earliest' || blockB === 0)) // only exception compare blocktag with number
 	) {
 		return 0;
 	}
-	if (blockA === 'earliest' || blockA === 0) {
-		// b !== a, thus a < b
-		return -1;
-	}
-	if (blockB === 'earliest' || blockB === 0) {
-		// b !== a, thus a > b
-		return 1;
-	}
-	if (blockA === 'latest' || blockA === 'safe') {
-		if (blockB === 'pending' || blockB === 'latest') {
+
```

Comment:
I think we can still compare `earliest` and  `0` with any tag or number. Beause `earliest` is a special tag that is always equal to `0`. And the following code is good enough for that:
```suggestion

	if (blockA === 'earliest' || blockA === 0) {
		// b !== a, thus a < b
		return -1;
	}
	if (blockB === 'earliest' || blockB === 0) {
		// b !== a, thus a > b
		return 1;
	}
```

However, some comments and tests would need to be updated accordingly.

Reply:
> I think we can still compare earliest and 0 with any tag or number. 
for earliest tag vs num:
what if user passes Number.MIN_VALUE , or run it on some chain with 0 blocks mined and latest/safe/finalized is also 0 (slim chances but still edge case that cannot be managed until we have block number for latest, safe, finalized,..etc).

Reply:
> we can still compare earliest and 0 with any tag or number.

1. we can compare `earliest` tag vs numbers `but with sanity check in place`: updated

2. 0 with `any tag` : I think not as in some edge cases as: if some one running local node for testing with block mining not set to time interval but set to some # of txs ( like geth insta mode ). In this case safe, latest etc block tags can be at 0 . so tags vs num case tag can be high or same and cant be compared until tag num is fetched first.

Reply:
Good catch!